src/main/java/org/jenkinsci/plugins/workflow/cps/global/UserDefinedGlobalVariable.java

```diff
@@ -66,6 +92,16 @@ public Object getValue(@Nonnull CpsScript script) throws Exception {
                we would also need to start calling LoadStepExecution.Replacer.
             */
             binding.setVariable(getName(), instance);
+
+            if (this.run != null) {
```

Comment:
Instead of threading the `Run` through the constructor, you could use `CpsScript.$build()`.